indico/modules/categories/client/js/ind-category-navigator.js

```diff
@@ -0,0 +1,33 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2024 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+
+import * as categoryNavigator from 'indico/react/components/category_navigator';
+
+import NavigateTo from './components/NavigateTo';
+
+customElements.define(
+  'ind-category-navigator',
+  class extends HTMLElement {
+    connectedCallback() {
```

Comment:
Should `disconnectedCallback` be handled as well (and unmount the react component)?

Reply:
Do you remove this widget from the page at any point?

Reply:
No, was just wondering.

FWIW when using it in other places, in particular for event creation (which is still a wtform and not using react directly but rather just to render individual widgets that happen to be implemented in react), we'd probably want to clean up (after closing the dialog the widget is no longer used, even though i guess there's no harm in leaving it mounted since the element it's mounted on no longer exists).

Reply:
In that case, we can add the unmount. Afaik, React does some subscriptions that won't be cleared when the node is removed.